shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1079 @@
+                picking, message=self.msg_store.need_confirmation()
+            )
+        self._handle_backorder(picking)
+        return self._response_for_select_document(
+            message=self.msg_store.transfer_done_success(picking)
+        )
+
+    def _handle_backorder(self, picking):
+        """This method handles backorders that could be created at picking confirm."""
+        backorders_before = picking.backorder_ids
+        picking._action_done()
+        backorders_after = picking.backorder_ids - backorders_before
+        # Remove user_id on the backorder, if any
+        backorders_after.user_id = False
+
+    def set_lot(
+        self, picking_id, selected_line_id, lot_name=None, expiration_date=None
+    ):
+        """Set lot and its expiration date
+
+        Input:
+            barcode: The barcode of a lot
+            expiration_date: The expiration_date
+
+        transitions:
+          - select_line: User clicked on back
+          - set_lot: Barcode not found. Ask user to create one from barcode
+          - set_lot: expiration_date has been set on the selected line
+          - set_lot: lot_it has been set on the selected line
+          - set_lot: Error: expiration_date is required
+          - set_quantity: User clicked on the confirm button
+        """
+        picking = self.env["stock.picking"].browse(picking_id)
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        message = self._check_picking_status(picking)
+        if message:
+            return self._response_for_set_lot(picking, selected_line, message=message)
+        if not selected_line.exists():
+            message = self.msg_store.record_not_found()
+            return self._response_for_set_lot(picking, selected_line, message=message)
+        search = self._actions_for("search")
+        if lot_name:
+            product = selected_line.product_id
+            lot = search.lot_from_scan(lot_name, products=product)
+            if not lot:
+                lot = self.env["stock.production.lot"].create(
+                    {
+                        "name": lot_name,
+                        "product_id": product.id,
+                        "company_id": self.env.company.id,
+                        "use_expiration_date": product.use_expiration_date,
+                    }
+                )
+            selected_line.lot_id = lot.id
+            selected_line._onchange_lot_id()
+        elif expiration_date:
+            selected_line.write({"expiration_date": expiration_date})
+            selected_line.lot_id.write({"expiration_date": expiration_date})
+        return self._response_for_set_lot(picking, selected_line)
+
+    def set_lot_confirm_action(self, picking_id, selected_line_id):
+        picking = self.env["stock.picking"].browse(picking_id)
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        message = self._check_expiry_date(selected_line)
```

Comment:
this is a check, is not meant to give you a msg IMO

Reply:
it's the same as _check_picking_status isn't it?